### ISSUE TITLE
Handle exceptions when in live-mode

### ISSUE THREAD

**Issue by bartsch** [tuhiproject/tuhi#123]
(Bamboo Slate small, used kete for testing)

- tuhi> enable_live ...
- turn device off
- tuhi> enable_live ... off

Doing the steps above results in an exception:
```
INFO: tuhi.ble: Disconnected
DEBUG: tuhi: FC:D5:9E:C4:DB:B0: disconnected
DEBUG: tuhi.dbus: Live mode stopped on Bamboo Slate for :1.336
DEBUG: tuhi.wacom: FC:D5:9E:C4:DB:B0: starting
DEBUG: tuhi.fw: TX Nordic --> ea / 02 / 03 00
INFO: tuhi.ble: FC:D5:9E:C4:DB:B0: Device is already disconnected
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/wacom.py", line 1213, in _run
    self._wacom_protocol.live_mode(args[1], args[2])
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/wacom.py", line 915, in live_mode
    self.width = self.get_dimensions('width')
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/wacom.py", line 615, in get_dimensions
    arguments=args)
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/wacom.py", line 380, in send_nordic_command_sync
    self.send_nordic_command(command, arguments)
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/wacom.py", line 331, in send_nordic_command
    chrc.write_value(data)
  File "/tmp/tuhi-install.d/lib/python3.7/site-packages/tuhi-0.1-py3.7.egg/tuhi/ble.py", line 64, in write_value
    return self.interface.WriteValue('(aya{sv})', data, {})
  File "/usr/lib/python3.7/site-packages/gi/overrides/Gio.py", line 354, in __call__
    None)
gi.repository.GLib.GError: g-io-error-quark: GDBus.Error:org.bluez.Error.Failed: Not connected (36)
```

(Looks like there're related issues. Though, I didn't knew where the trace fits best.)

**Comment by ckilimci** [comment]
Hi all,

I also reproduce this bug. I can help but I do not know where to start. Can you give me a hint?

**Comment by whot** [comment]
First, stop tuhi, note the current time, restart tuhi. Reproduce the issue.

Then go to  `$HOME/.local/share/tuhi/session-logs/tuhi-<currenttime.log>`. That file has all the output you've seen on the terminal. It also includes a line that says `sessionid: 65b27aeff2744ec0876f689f1f01be4f-2` or similar. The first part is random, the `-2` in this example increments with each device connection.

Find the connection where it all crashed, then go to`$HOME/.local/share/tuhi/<bluetooth addr>/raw`. grep for the matching sessionid in those files and attach that one here, together with the session log itself. That should give us some hints at what's happening.

**Comment by whot** [comment]
@ckilimci definitely update to #210 before any testing, live-mode was accidentally broken on master, sorry.

**Comment by ckilimci** [comment]
Thank you. I could not update yesterday because I was not able to reproduce. Let me try again.

**Comment by whot** [comment]
Let's close this, it's been almost 3 years. please re-open if it's still an issue
